src/appleseed/renderer/kernel/lighting/pt/ptlightingengine.cpp

```diff
@@ -710,4 +710,61 @@ ILightingEngine* PTLightingEngineFactory::create()
     return new PTLightingEngine(m_light_sampler, m_params);
 }
 
+Dictionary PTLightingEngineFactory::get_params_metadata()
+{
+    Dictionary metadata;
+    add_common_params_metadata(metadata);
+
+    metadata.dictionaries().insert(
+        "enable_dl",
+        Dictionary()
+            .insert("type", "bool")
+            .insert("help", "Enable direct lighting"));
+
+    metadata.dictionaries().insert(
+        "enable_caustics",
+        Dictionary()
+            .insert("type", "bool")
+            .insert("default", "false")
+            .insert("help", "Enable caustics"));
+
+    metadata.dictionaries().insert(
+        "max_path_lenght",
+        Dictionary()
+            .insert("type", "int")
+            .insert("help", "Maximum number of light bounces"));
+
+    metadata.dictionaries().insert(
+        "rr_min_path_length",
+        Dictionary()
+            .insert("type", "int")
+            .insert("help", "TODO: Not documented yet"));
+
+    metadata.dictionaries().insert(
+        "dl_light_samples",
+        Dictionary()
+            .insert("type", "int")
+            .insert("help", "Number of samples used to estimate direct lighting"));
+
+    metadata.dictionaries().insert(
+        "ibl_env_samples",
+        Dictionary()
+            .insert("type", "int")
+            .insert("help", "Number of samples used to estimate environment lighting"));
+
+    metadata.dictionaries().insert(
```

Comment:
Next event estimation should pretty much always be enabled. It is kept for educational purposes only. I advise against exposing it.